File: DataBindingHandlerJava/app/src/main/java/com/websarva/wings/android/databindinghandlerjava/ui/MainHandlers.java
package com.websarva.wings.android.databindinghandlerjava.ui;

import com.websarva.wings.android.databindinghandlerjava.databinding.ActivityMainBinding;

public class MainHandlers {
	MainViewModel _mainViewModel;
	ActivityMainBinding _activityMainBinding;

	public MainHandlers(MainViewModel mainViewModel, ActivityMainBinding activityMainBinding) {
		_mainViewModel = mainViewModel;
		_activityMainBinding = activityMainBinding;
	}

	public void onBtnGenerateClick() {
		_mainViewModel.generateNewRectangle();
		_activityMainBinding.setRectangle(_mainViewModel.getRectangle());
	}
}

File: DataBindingHandlerKotlin/app/src/main/java/com/websarva/wings/android/databindinghandlerkotlin/ui/MainHandlers.kt
package com.websarva.wings.android.databindinghandlerkotlin.ui

import com.websarva.wings.android.databindinghandlerkotlin.databinding.ActivityMainBinding

class MainHandlers(private val _mainViewModel: MainViewModel, private val _activityMainBinding: ActivityMainBinding) {
	fun onBtnGenerateClick() {
		_mainViewModel.generateNewRectangle()
		_activityMainBinding.rectangle = _mainViewModel.rectangle
	}
}
